Test/com/nicoles_change_maker/test/ChangeMakerTest.java

```diff
@@ -0,0 +1,235 @@
+package com.nicoles_change_maker.test;
+import com.nicoles_change_maker.ChangeMaker;
+import org.junit.*;
+import java.util.HashMap;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Created by bears8yourface on 1/5/16.
+ */
+
+public class ChangeMakerTest {
+    ChangeMaker changeMaker;
+    HashMap expectedChange = new HashMap<String,Integer>();
+    HashMap change = new HashMap<String,Integer>();
+
+    @Before
+    public void setUp() {
+        changeMaker = new ChangeMaker();
+    }
+
+    @Test
+    public void testInputTakesPositiveInteger() {
+        changeMaker.makeChange(0);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testInputRejectsNegativeInteger() {
+        changeMaker.makeChange(-2);
+    }
+
+    @Test
+    public void testChangeForZeroDollars() {
+        change = changeMaker.makeChange(0);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForOneCent() {
+        expectedChange.put("P", 1);
+        change = changeMaker.makeChange(.01);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForTwoCents() {
+        expectedChange.put("P", 2);
+        change = changeMaker.makeChange(.02);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForThreeCents() {
+        expectedChange.put("P", 3);
+        change = changeMaker.makeChange(.03);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForFourCents() {
+        expectedChange.put("P", 4);
+        change = changeMaker.makeChange(.04);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForFiveCents(){
+        expectedChange.put("N", 1);
+        change = changeMaker.makeChange(.05);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSixCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 1);
+        change = changeMaker.makeChange(.06);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForSevenCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 2);
+        change = changeMaker.makeChange(.07);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForEightCents(){
+        expectedChange.put("N", 1);
+        expectedChange.put("P", 3);
+        change = changeMaker.makeChange(.08);
+        assertEquals(expectedChange, change);
+    }
+
+    @Test
+    public void testChangeForNineCents(){
```

Comment:
Were these tests (one nickel, many pennies) driving your production code, or just verifying that it worked as you expected?